File: app/src/main/java/com/example/weatherapplication/CityFragment/CityHourlyWeather.java
package com.example.weatherapplication.CityFragment;


import com.example.weatherapplication.WeatherImage;

public class CityHourlyWeather {
    private long dataReceivingTime;
    private double temperature;
    private String description;
    private int imagePath;

    CityHourlyWeather(long dataReceivingTime, double temperature,String description,long sunrise, long sunset){
        this.dataReceivingTime = dataReceivingTime;
        this.temperature = temperature;
        this.description = description;
        setImage(sunrise,sunset);
    }


    public void setImage(long sunrise, long sunset){
        if (dataReceivingTime >= sunrise && dataReceivingTime<=sunset)
            this.imagePath = WeatherImage.getImageRecourse(description,"day",temperature);
        else
            this.imagePath = WeatherImage.getImageRecourse(description,"night",temperature);
    }

    public long getDataReceivingTime() {
        return dataReceivingTime;
    }

    public double getTemperature() {
        return temperature;
    }

    public int getImagePath() {
        return imagePath;
    }

}

File: app/src/main/java/com/amirreza/domain/entity/OneCallWeatherEntitys/CityHourlyWeather.kt
package com.amirreza.domain.entity.OneCallWeatherEntitys

import android.R.attr.description
import com.amirreza.domain.entity.WeatherImage
import com.google.gson.annotations.SerializedName


data class CityHourlyWeather(
    val clouds: Int,
    @SerializedName("dew_point")
    val AtmosphericTemperature: Double,

    @SerializedName("dt")
    val requestTime: Int,

    @SerializedName("feels_like")
    val feelsLike: Double,
    val humidity: Int,
    val pop: Int,
    val pressure: Int,
    val temp: Double,
    val uvi: Double,
    val visibility: Int,

    @SerializedName("weather")
    val weatherDescription: List<WeatherDescription>,
    val wind_deg: Int,
    val wind_gust: Double,
    val wind_speed: Double,

    var imagePath:Int?
){
    fun setImage(sunrise: Long, sunset: Long) {
        imagePath = if (requestTime in sunrise..sunset)
            WeatherImage.getImageRecourse(weatherDescription[0].description, "day", temp)
        else WeatherImage.getImageRecourse(weatherDescription[0].description, "night", temp)
    }
}